lib/js_render/renderer.rb

```diff
@@ -38,7 +42,8 @@ def generate_html
       server_code = <<-JS
         (function () {
           var serverStr = typeof #{func_name} === 'function' ? #{func_name}(#{@json_data}) : '';
-          return '<span id="#{@uuid}">' + serverStr + '</span>';
+          var consoleReplayStr = #{CONSOLE_REPLAY};
+          return '<span id="#{@uuid}">' + serverStr + '</span>' + consoleReplayStr;
```

Comment:
I think it might make more sense structurally to move this into `render_component`. Right now render component gets the `server_html` and the `client_script` then concats and returns them. I think we can then add the replay script as the third piece in there.

Reply:
In order for the replay to access a  `console` object with a `history` property filled with actions attempted by the `serverStr` code, it needs to be run in the same context as `serverStr`. This was definitely a tricky piece, so any ideas you have on how to get around it would be appreciated.

Reply:
ahh I see what you mean. that makes sense, I missed that on my first review